packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -199,20 +199,29 @@ export class IndexedDbPersistence implements Persistence {
   private updateClientMetadataAndTryBecomePrimary(): Promise<void> {
     return this.simpleDb.runTransaction('readwrite', ALL_STORES, txn => {
       const metadataStore = clientMetadataStore(txn);
-      metadataStore.put(
-        new DbClientMetadata(this.clientId, Date.now(), this.inForeground)
-      );
+      return metadataStore
+        .put(new DbClientMetadata(this.clientId, Date.now(), this.inForeground))
+        .next(() =>
+          this.canActAsPrimary(txn).next(canActAsPrimary => {
```

Comment:
Can you move the paren on line 217 to after `this.canActAsPrimary(txn)` so this looks nicer / is less nested?

Also, just to confirm, nothing actually changed in this method, right? It's just syntax?